### PR TITLE
add tests for fq2bam

### PR DESCRIPTION
**Pull request by sateeshperi**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -644,6 +644,12 @@ jobs:
             path: modules/nf-core/parabricks/fq2bammeth
           - profile: singularity
             path: modules/nf-core/parabricks/fq2bammeth
+          - profile: conda
```
Comment (GallVp) [inline .github/workflows/test.yml:647]: Removed these from test.

File: modules/nf-core/parabricks/fq2bam/meta.yml [hunk 2]
```diff
@@ -45,6 +40,11 @@ input:
         type: file
         description: reference BWA index
         pattern: "*.{amb,ann,bwt,pac,sa}"
+  - - interval_file:
```
Comment (GallVp) [inline modules/nf-core/parabricks/fq2bam/meta.yml:43]: Used `nf-core --fix lint`

File: modules/nf-core/parabricks/fq2bam/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+    test("sarscov2 - fastq - pe") {
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:false ], // meta map
+                    [
+                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
```
Comment (GallVp) [inline modules/nf-core/parabricks/fq2bam/tests/main.nf.test:113]: Added a test for PE data

**Review by GallVp — APPROVED** [review]

**Comment by GallVp** [comment]
Hi @sateeshperi 

I'm happy with it and confirm that it runs on AWS/batch.

Please have a look and merge.

File: modules/nf-core/parabricks/fq2bam/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+                input[0] = Channel.of([
+                    [ id:'test' ], // meta map
+                    file('https://github.com/nf-core/test-datasets/raw/methylseq/reference/genome.fa', checkIfExists: true)
+                ])
+                """
+            }
+        }
+
+        run("BWA_INDEX", alias: 'BWA_INDEX_PE') {
+            script "../../../bwa/index/main.nf"
```
Comment (sateeshperi) [inline modules/nf-core/parabricks/fq2bam/tests/main.nf.test:28]: why a separate setup for PE ? it cna be reused right ?

Comment (GallVp) [reply modules/nf-core/parabricks/fq2bam/tests/main.nf.test:28]: Because PE is using a different reference!

Comment (sateeshperi) [reply modules/nf-core/parabricks/fq2bam/tests/main.nf.test:28]: of course....plz ignore :)

**Pull request merged**